content/howto/extensibility/how-to-create-a-native-pluggable-widget.md

```diff
@@ -0,0 +1,1344 @@
+       } else if (headerCaption) {
+           return view;
+       }
+
+       return null;
+   };
+   ```
+
+3. Change the **renderIcon** method to prevent an icon from being rendered when the group box isn't collapsible by adding a guard that returns null:
+
+   ```tsx
+   private renderIcon = (): ReactNode => {
+       const { collapsible, collapseIcon, expandIcon } = this.props;
+
+       if (!collapsible) {
+           return null;
+       }
+
+       if (this.state.collapsed) {
+           return expandIcon ? expandIcon : <Text style={this.styles.headerContent}>+</Text>;
+       }
+
+       return collapseIcon ? collapseIcon : <Text style={this.styles.headerContent}>-</Text>;
+   };
+   ```
+
+4. Change the **render** method to deal correctly with the situation where there is no header nor content to be rendered by returning null when this situation occurs:
+
+   ```tsx
+   render(): ReactNode {
+       const renderedHeader = this.renderHeader();
+       const renderedContent = this.renderContent();
+
+       if (!renderedHeader && !renderedContent) {
+           return null;
+       }
+
+       return (
+           <View style={this.styles.container}>
+               {renderedHeader}
+               {renderedContent}
+           </View>
+       );
+   }
+   ```
+
+The last thing we need to do is changing the container component so that the properties get passed to the display component.
+
+1. Navigate to the container component.
+
+2. Change the **render** method to supply the display component with the information from the collapsible prop that the container component receives:
+
+   ```tsx
+       render(): ReactNode {
+           const { collapsible, collapseIcon, expandIcon, content, headerCaption, style } = this.props;
+
+           const isCollapsible = collapsible !== "no";
+
+           const props: WrappedGroupBoxProps = {
+               collapsible: isCollapsible,
+               headerCaption,
+               collapseIcon: this.renderIcon(collapseIcon, defaultCollapseIconGlyph),
+               expandIcon: this.renderIcon(expandIcon, defaultExpandIconGlyph),
+               style
+           };
+
+           if (collapsible) {
+               props.startCollapsed = collapsible === "yesStartCollapsed";
+           }
+
+           return <WrappedGroupBox {...props}>{content}</WrappedGroupBox>;
+       }
+   ```
+
+### 3.6 Adding custom default style
+
+Although we have an extensively featured group box, we can still improve the styling of the widget. Right now, we apply very basic styling that's baked into our widget code. We can let the Mendix Client override this default styling by defining a custom style for our widget in Atlas UI.
+
+Before we are going to define a custom style, we want to align the widget mpk file name and the widget's deployment folder structure with other Mendix widgets.
+
+1. Navigate to **package.json**
+
+2. Change the **packagePath** value from "mendix" to "com.mendix.widget.native":
```

Comment:
should have done this with the name of the generator, company name is com.mendix.widget.native

this saves you from explain this hole step

Reply:
Changed it and removed a lot of steps.